crates/solver/src/liquidity/zeroex.rs

```diff
@@ -30,18 +30,24 @@ impl ZeroExLiquidity {
             },
         ];
 
-        let zeroex_orders =
-            futures::future::try_join_all(queries.iter().map(|query| self.api.get_orders(query)))
-                .await?;
+        let zeroex_orders_results =
+            futures::future::join_all(queries.iter().map(|query| self.api.get_orders(query))).await;
+        let zeroex_orders = zeroex_orders_results
+            .into_iter()
+            .flat_map(|result| match result {
+                Ok(order_record_vec) => order_record_vec,
+                Err(err) => {
+                    tracing::warn!("ZeroExResponse error during liqudity fetching: {}", err);
```

Comment:
In general, its good to have metrics for measuring errors that we don't alert on.

That being said, I think we already count the number of 0x liquidity orders with another metric, so we can use that to see if we aren't getting any 0x liquidity through.

Also, are these errors intermittent? Is there some underlying issue that is causing them to happen (such as being rate limited? - if so maybe we can consider using the new `RateLimiter` that @MartinquaXD introduced?).

Reply:
> That being said, I think we already count the number of 0x liquidity orders with another metric, so we can use that to see if we aren't getting any 0x liquidity through.

There is currently only a metric for the number of `LimitOrder` liquidity items. At the moment we only get `LimitOrder` liquidity from 0x but this might change in the future. This metric also only shows the number of **usable** 0x orders we got. I believe we could theoretically get 0 usable 0x orders without errors while collecting liquidity.

> if so maybe we can consider using the new RateLimiter

The underlying error is some generic `reqwest::Error` which happened during `RequestBuilder::send()`. The docs say the following about that:
```rust
    /// # Errors
    ///
    /// This method fails if there was an error while sending request,
    /// redirect loop was detected or redirect limit was exhausted.
    pub fn execute(
        &self,
        request: Request,
    ) -> impl Future<Output = Result<Response, crate::Error>> {
```
So I don't think there is anything we can do about that. To make use of the `RateLimiter` we need to actually get a response with status code `429`.

Reply:
Thanks, Martin, for the investigation!